### ISSUE TITLE
Feature Request: Multiple Data Directories or Workspaces etc

### ISSUE THREAD

**Issue by Philiphlop** [fabiospampinato/notable#60]
The reason behind the request is separation of data for different purposes. My work has a corporate sync (box.com), and I use a personal one (dropbox) for my own notes. This means my work stuff is synced only to work systems, not my personal desktop at home, which is what I use Dropbox on. 

I would like to be able to combine them both into the same app front end, but separate at the data location, so I can be compliant with data locality. I have used VSCode in the past for this, doing different folders/workspaces, in the same sidebar, for one place to go for notes. But VSCode is a poor notes app.

**Comment by Philiphlop** [comment]
Yes it would, that or having multiple instances open with a data directory. Whether that's done via the app, or I put in the data directory as an argument in the shortcut/launch command. The notes are for different uses, hence different protections around the data, so don't actually need to mix in the UI.

**Comment by timkinnane** [comment]
I was hoping to use Notable as a UI for content managing my static site, which compiles most pages from markdown. I tried changing the working directory to my `/docs` path, but it seems to require all files are created in the "notes" subpath and couldn't read the existing markdown files. This could be a seperate issue, but definitely related. When changing path, could the behaviour be changed to allow top-level files?

**Comment by Fastidious** [comment]
@timkinnane depending on which static generator you use, Notable might (will?) break things for you, as I believe it modifies the front matter. For example, I use `tags` on Hugo, which Notable also uses on its own way. I think you want Notable to be a markdown editor, but it is not.

**Comment by fabiospampinato** [comment]
@timkinnane yeah as @Fastidious mentioned this might not be such a good idea. There's been some discussion about this in the following issues, generally I don't think a change in the structure will happen:

- https://github.com/fabiospampinato/notable/issues/220
- https://github.com/fabiospampinato/notable/issues/218
- https://github.com/fabiospampinato/notable/issues/54

**Comment by fabiospampinato** [comment]
Let's wait until a proper command panel #338 is implement first.

**Comment by fabiospampinato** [comment]
> Would instead a much better/faster way of switching between data directories be enough for you?

That's been implemented in v1.6, there's now a shortcut for changing data directory, and when doing so the window gets refreshed rather than fully reloaded, which is much faster.

**Comment by djthomson** [comment]
_I appreciate that this is being looked into and improvements are being made, however accessing different data directories is still relatively clunky and slow. I created a duplicate issue by mistake, apologies. However, copying here for the purposes of discussion:_

I would love to be able to define multiple data directories and ideally access these simultaneously.

I need to use different directories in order to access resources on remote/encrypted drives and to manage permissions settings effectively. Sadly, the time it takes to change the active data directory manually each time is prohibitive as I need to work on files from several different paths and switch between them frequently.

If this could be implemented, it would make it much easier for users to provide their own solutions for remote sync, encryption etc.; it would greatly improve convenience for users using remote or removeable devices and would give users better control over how their data can be stored, even on their local filesystem.

Perhaps Notable could retain a default/master data directory, with additional paths/directories accessible either through:

1.     The sidebar, similar to a top-level or special tag (which would be my preference)
2.     Different instances of Notable
3.     Shortcuts to select/switch the active data directory from a set of favourites

Thanks for your work on this amazing application!

**Comment by fabiospampinato** [comment]
> The sidebar, similar to a top-level or special tag (which would be my preference)

Sounds reasonable, my best guess is that at some point this will be supported, but it's not a priority right now.

> Different instances of Notable

Definitely.

> Shortcuts to select/switch the active data directory from a set of favourites

Totally, I need to implement a command palette first though #338.

**Comment by derfrankie** [comment]
Hi,
by reading the posts I see that it's a similar "problem" what I have.

1) I switch regularly between data folders as I keep small Notable folders in each projects - A subfolder in the Menu similar to "Recent Files" in other apps would be a great improvement

2) Is there a way to put a "shortcut file" in the base of the data folder, which would open Notable with that data directory? As I said before I have plenty of small Notable folder - for each project one - and it would help immensly if I could just "open" Notable fron the finder with the right data folder set :)

**Comment by fabiospampinato** [comment]
@derfrankie If you put all your data directories in a common parent directory you should be able to switch them pretty quickly via `CtrlOrCmd+Shift+X`, as by default the dialog opens in the parent directory.

> Is there a way to put a "shortcut file" in the base of the data folder, which would open Notable with that data directory? 

In theory no, but the data directory to open is the value of the "cwd" property in `~/.notable.json`, so if you really wanted to you could make a script which updates that field and then opens Notable.

**Comment by bcardiff** [comment]
What do you think about storing in `.notable.json` a small list of recent data directories? So an `Open Recent...` menu can be added listing them. It could be a good alternative for switching between workspaces.

The list can be built by the app and the user is able to override it.

**Comment by fabiospampinato** [comment]
> What do you think about storing in .notable.json a small list of recent data directories? So an Open Recent... menu can be added listing them. It could be a good alternative for switching between workspaces.

Yes something like that will be implemented.

**Comment by darkphase** [comment]
Tags may be grouped in domains and attached with "type" metadata may solve all hierarchical arrangement problems...
For example, let tag domains be; System, Security, User
The tags we (users) created may fall into User domain, Data Folders, Usage Statistics, Creation Time fall into System domain etc...

**Comment by fabiospampinato** [comment]
I'd be interested to hearing some of your feedback on the following options:

1. The app could handle multiple data directory and have pretty much the same interface as when working with a single data directory.
2. The app could implement tabbing and have a different data directory under each tab.
3. We could add support for opening multiple windows and switching to data directories very quickly.

Which option would you guys prefer? IMHO option 3 in some ways seems the more flexible and definitely the easier to implement.

**Comment by derfrankie** [comment]
Option 3. Multiple windows would work best for me, as I wanted to keep the notes as separated as possible. 

In the meantime I use a lot the workspace feature of vscode, where you can store your folder config etc in a file and just start from there.

**Comment by wamserma** [comment]
Also voting for Option 3 with a list of favourites/recent notebooks for quick access.

**Comment by JannieT** [comment]
Option 3
